homeassistant/components/dsmr/sensor.py

```diff
@@ -460,7 +460,6 @@ def rename_old_gas_to_mbus(
                         ent_reg.async_update_entity(
                             entity.entity_id,
                             new_unique_id=mbus_device_id,
-                            device_id=mbus_device_id,
```

Comment:
This is the serial number of the device, not a device registry id.

Reply:
Looks correct. I think this legacy correction could possible be removed soon.